### PR TITLE
[T5] Fix init in TF and Flax for pretraining

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Fixes #16749

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: src/transformers/models/t5/modeling_t5.py [hunk 1]
```diff
@@ -768,6 +768,9 @@ def _init_weights(self, module):
             # Mesh TensorFlow embeddings initialization
             # See https://github.com/tensorflow/mesh/blob/fa19d69eafc9a482aff0b59ddd96b025c0cb207d/mesh_tensorflow/layers.py#L1624
             module.shared.weight.data.normal_(mean=0.0, std=factor * 1.0)
+            if hasattr(module.lm_head) and not self.config.tie_word_embeddings:
+                module.lm_head.weight.data.normal_(mean=0.0, std=factor * 1.0)
+            module.shared.weight.data.normal_(mean=0.0, std=factor * 1.0)
```
Comment (patrickvonplaten) [inline src/transformers/models/t5/modeling_t5.py:773]:
```suggestion
```

File: src/transformers/modeling_tf_utils.py [hunk 2]
```diff
@@ -1975,6 +1975,7 @@ def build(self, input_shape):
         Build shared token embedding layer Shared weights logic adapted from
         https://github.com/tensorflow/models/blob/a009f4fb9d2fc4949e32192a944688925ef78659/official/transformer/v2/embedding_layer.py#L24
         """
+
```
Comment (sgugger) [inline src/transformers/modeling_tf_utils.py:1978]: Maybe leave it as is?
```suggestion
```

**Review by sgugger — APPROVED** [review]
Thanks for fixing those!

File: src/transformers/modeling_tf_utils.py (hunk 2 above)
Comment (patrickvonplaten) [reply src/transformers/modeling_tf_utils.py:1978]: ups sorry

**Review by patil-suraj — APPROVED** [review]
Thank you for fixing this!

**Pull request merged**